### ISSUE TITLE
Website needs to be opened - mentionbb.com 

### ISSUE THREAD

**Issue by slaweally** [mentionbb/mentionbb#6]
Friends, I am not willing to leave such a beautiful project in the middle :) How about a small update every day? At the very least, I think it is enough to make it clear that the project is being taken care of. 

At least we are not updating, at least let's open the forum :)

**Comment by par274** [comment]
Thank you.

Actually, the project was not abandoned. I am making small updates but I haven't been able to do much with it due to my work these days. Currently, many of its functions work as they should, but there are many shortcomings in the admin panel and I could not spare any time for the Install interface.

I will activate the website as soon as possible.

**Comment by slaweally** [comment]
If you want, let's do it like this, when you open the site, let's test it in free time for review and development, let's take note of the deficiencies and errors, let's go step by step and update it.

As a tester, I have a lot of review time in my free time, but how much time do you have to update 😁

**Comment by par274** [comment]
If something is broken or needs to be replaced, it needs to be updated immediately, so that time can be created. I'll post it here when I open the site.

Until then, you can pull the project to local. The project actually had a Docker* repo, but it's older than mine. So I need to open that too. :D

**Comment by slaweally** [comment]
It seems that you did not discard the project but abandoned it 😁😁😁
Please can I ask you to take over the project as soon as possible so that we can continue this work more seriously and with greater dedication 😁😁

**Comment by par274** [comment]
No no, the project has not been abandoned. I try to make updates occasionally.

I wrote Doctrine above, it was to be Docker. This is proof that I don't have time. 😁

**Comment by slaweally** [comment]
Okay, now let's be more dedicated, more hardworking and let's start making this work 🫡

**Comment by par274** [comment]
If you are going to use it with Docker, I upload repo.

https://github.com/mentionbb/mentionbb-dockerized

**Comment by slaweally** [comment]
> Eğer Docker ile kullanacaksanız, repo yükler.
> 
> https://github.com/mentionbb/mentionbb-dockerized

In fact, it should be a very simple installler for the end user, not for us, we are aware of this, but I think this is one of the most important ones, isn't it? 

Because unzip it from the zip, enter the sql information, create admin and log in to the system, I think the first step is to create such a simplified installation package :) 

I will examine it a little personally, but as I said, this should be one of the most important ones, I think we can say that it should be done first.

**Comment by par274** [comment]
> > Eğer Docker ile kullanacaksanız, repo yükler.
> > https://github.com/mentionbb/mentionbb-dockerized
> 
> In fact, it should be a very simple installler for the end user, not for us, we are aware of this, but I think this is one of the most important ones, isn't it?
> 
> Because unzip it from the zip, enter the sql information, create admin and log in to the system, I think the first step is to create such a simplified installation package :)
> 
> I will examine it a little personally, but as I said, this should be one of the most important ones, I think we can say that it should be done first.

This is true. In fact, the project has an ongoing Install interface. However, sometimes the core files in the project can change and therefore the Install to be done at this stage can cause me more trouble.

**Comment by slaweally** [comment]
Yes, you may be right about this, but if we look at things from the point of view of the end user, not from our point of view, it offers a very negative pre impression to the user.

**Comment by par274** [comment]
Actually, everything is ready for this. Table layouts etc. just need to be directed to the install wizard correctly. I will update this issue as I find time.

https://github.com/mentionbb/mentionbb/tree/master/src/App/Install

Theoretically, it should be ready by the end of the year.

**Comment by ardaabi1** [comment]
For the forum site's theme, it would be great if the design prioritizes user convenience, while also ensuring that moderators and administrators can easily manage tasks within the admin panel. These are just my suggestions; I don’t have much expertise in PHP, Docker, or similar technologies. Ensuring compatibility with existing sites is crucial for the project's continuity. Having an install page that automatically sets up the data would also be beneficial.

**Comment by par274** [comment]
> For the forum site's theme, it would be great if the design prioritizes user convenience, while also ensuring that moderators and administrators can easily manage tasks within the admin panel. These are just my suggestions; I don’t have much expertise in PHP, Docker, or similar technologies. Ensuring compatibility with existing sites is crucial for the project's continuity. Having an install page that automatically sets up the data would also be beneficial.

I would like to state that I have a more modern approach in terms of design. I have moved away from the classic, vBulletin, XenForo sketch and have been inspired by more modern projects such as Discourse and Flarum.

Actually, I need to focus on Install but I don't have much time, but that doesn't mean it will never be done. There are a lot of negative comments about this and they are right. I will increase my work on the Install interface.

![resim](https://github.com/user-attachments/assets/87c17d39-285b-4fbf-b6d1-a8e61bd2891e)

Additionally, we have an admin panel. From here you can manage members, themes, plugins and even templates.

**Comment by ardaabi1** [comment]
I will set up the site according to the instructions and take a look at it. Is there additional language support?

**Comment by par274** [comment]
> I will set up the site according to the instructions and take a look at it. Is there additional language support?

Currently Turkish and English support only. Multilingual support is available but translation into other languages ​​is required. You can find the files in src/languages.

**Comment by ardaabi1** [comment]
Thank you, I will provide my comments on this topic from a user perspective after a detailed review over time.

**Comment by ardaabi1** [comment]
Frankly, the feedback is as follows. The database config file is very complex and cannot be set up properly; I haven’t been able to find where I am making a mistake, but I am continuing to try to install it. If there is an explanation video, etc., it would be very helpful. (This is regarding cPanel installation)

**Comment by par274** [comment]
> Frankly, the feedback is as follows. The database config file is very complex and cannot be set up properly; I haven’t been able to find where I am making a mistake, but I am continuing to try to install it. If there is an explanation video, etc., it would be very helpful. (This is regarding cPanel installation)

Actually, it would be more useful to open a separate issue for this. In this issue, the other friend said that the website should be opened. The two issues are very unrelated.

The DbConfig.php file is currently a complex structure because it contains all the adapters. The reason for this is to keep the config order of the adapters. Actually, this file is a file that will be created automatically by the Install Wizard. Right now, it just needs to be done manually.

If you are installing with Cpanel, you are most likely using MYSQL_PDO. In other words, the part you need to set is the username and password in the "DB_Params" section and the dbname and host under the "DB_Params_PDO_MySQL" heading.

Host is generally "localhost".

**Comment by slaweally** [comment]
Dear friends, please let's not use the topic in a brutal way, I opened this topic, everyone should go to their own topic :))))

Just kidding, now let me give additional information for the part that the friend said, I did a rough installation yesterday, but it is necessary to edit everything manually, so I tried a little bit and got bored and left it :) 
To be honest, it seems that there should be an installation package for a while, at least for those at the basic level. 

I got stuck on the Nginx configuration part, I didn't dwell on it too much to be honest :)

**Comment by par274** [comment]
> Dear friends, please let's not use the topic in a brutal way, I opened this topic, everyone should go to their own topic :))))
> 
> Just kidding, now let me give additional information for the part that the friend said, I did a rough installation yesterday, but it is necessary to edit everything manually, so I tried a little bit and got bored and left it :) To be honest, it seems that there should be an installation package for a while, at least for those at the basic level.
> 
> I got stuck on the Nginx configuration part, I didn't dwell on it too much to be honest :)

Actually, the nginx configuration file is related to the server, not the software. But I still gave an example of this.

For example, the Install interface of high-level software such as vBulletin and XenForo does not say anything about the Nginx installation. In fact, it cannot be intervened because, as I said, it is related to the server.

Additionally, MentionBB also runs smoothly on Apache. Nginx is not a requirement.

In fact, I added the Docker version so that people don't have to deal with these. It provides a better testing environment. Of course, some Docker knowledge is required for this.

It is easy to install the software without Docker, but it requires settings on the Nginx server. Or it can be used without Nginx on Apache without any settings, except for the database settings of course.

**Comment by slaweally** [comment]
> > Dear friends, please let's not use the topic in a brutal way, I opened this topic, everyone should go to their own topic :))))
> > Just kidding, now let me give additional information for the part that the friend said, I did a rough installation yesterday, but it is necessary to edit everything manually, so I tried a little bit and got bored and left it :) To be honest, it seems that there should be an installation package for a while, at least for those at the basic level.
> > I got stuck on the Nginx configuration part, I didn't dwell on it too much to be honest :)
> 
> Actually, the nginx configuration file is related to the server, not the software. But I still gave an example of this.
> 
> For example, the Install interface of high-level software such as vBulletin and XenForo does not say anything about the Nginx installation. In fact, it cannot be intervened because, as I said, it is related to the server.
> 
> Additionally, MentionBB also runs smoothly on Apache. Nginx is not a requirement.



No no no, it's not a problem, it's all about me, you know nginx is like a nice lady who asks for special configurations so I didn't dwell on it too much.

But this installation part that the other friend mentioned is really vital, I'll even give you a little suggestion for the installer in the installation part if you want to take a note;
We can show a small warning so that the installer automatically recognizes the server server and redirects accordingly,
For example: Server type: Nginx, see special configuration file for installation (nginx-install.txt)

Server type: Aphache, You do not need to take any action, you can continue with the installation.
Server type: Litespeed, No action required. 

You can also use these suggestions, of course this is my own opinion :)

**Comment by par274** [comment]
> > > Dear friends, please let's not use the topic in a brutal way, I opened this topic, everyone should go to their own topic :))))
> > > Just kidding, now let me give additional information for the part that the friend said, I did a rough installation yesterday, but it is necessary to edit everything manually, so I tried a little bit and got bored and left it :) To be honest, it seems that there should be an installation package for a while, at least for those at the basic level.
> > > I got stuck on the Nginx configuration part, I didn't dwell on it too much to be honest :)
> > 
> > 
> > Actually, the nginx configuration file is related to the server, not the software. But I still gave an example of this.
> > For example, the Install interface of high-level software such as vBulletin and XenForo does not say anything about the Nginx installation. In fact, it cannot be intervened because, as I said, it is related to the server.
> > Additionally, MentionBB also runs smoothly on Apache. Nginx is not a requirement.
> 
> No no no, it's not a problem, it's all about me, you know nginx is like a nice lady who asks for special configurations so I didn't dwell on it too much.
> 
> But this installation part that the other friend mentioned is really vital, I'll even give you a little suggestion for the installer in the installation part if you want to take a note; We can show a small warning so that the installer automatically recognizes the server server and redirects accordingly, For example: Server type: Nginx, see special configuration file for installation (nginx-install.txt)
> 
> Server type: Aphache, You do not need to take any action, you can continue with the installation. Server type: Litespeed, No action required.
> 
> You can also use these suggestions, of course this is my own opinion :)

Ah yes, that makes a lot of sense actually. I'll add it to my notes. Actually, Apache doesn't need any settings, Nginx is a bit of a delicate tool.

Actually, I have a lot of plans for the install; ai language translation, downloading all the necessary files from github, usage documentation. But right now my goal is to produce a working install. I am as aware as you are of the high level complexity of even just the DbConfig.php file for the end user.

![resim](https://github.com/user-attachments/assets/3ce7c952-58b2-4ae1-a3bb-d37c54cbc213)


Speaking of documentation, I also need to prepare a tutorial for making plugins and themes. There's a lot of work to be done.

**Comment by slaweally** [comment]
Yes, it's obvious how much work there is, but it's all something that will happen over time. it's a step by step, gradual progress. but first and foremost to attract the end user, the first step is a very simple installed system. 
download as zip -> enter installation info -> forum is ready. 
Once this part is done there will be plenty of time for other background work. new developments, additions and edits can already be done based on user comments and feedback. 
Again, as I said, opening the website is a must :)

**Comment by par274** [comment]
Demo is open now.

https://mentionbb.com/

**Comment by slaweally** [comment]
> Demo is open now.
> 
> https://mentionbb.com/

Hello again
We can actually do something like this, if you accept, I have prepared a portal introduction for Mentionbb, you can look at it here; https://github.com/slaweallx/mentionbb/blob/master/mention.zip

Mentionbb.com welcomed us here at the entrance, followed by the Forum (active for the real community) not the demo
Download directory (All versions)
You can divide it into sections like this, this is just an idea.

**Comment by par274** [comment]
The idea is good but I guess this is a draft?

No need for a separate project, we can do this as a addon. Let me set up the addon files, you can do it with HTML and send a pull-request. Do you know how to use Twig Template?

**Comment by slaweally** [comment]
Aaa no no no, I made it as a portal just for the entrance, let the HTML stay, for example, if you look at smf mybb, I thought as a simple portal to the entrance like them, I thought we would welcome there. But it can also be done as an add-on, of course, it can be made active closed from the panel as a portal, but I have arranged and prepared it only for mentionbb specially, so it is not for everyone to use it.

**Comment by par274** [comment]
It can be done as an add-on, but making it so that the end user can turn it on and off doesn't make sense right now.

Frankly, your idea is great, but the design is not worthy of Mention. You can make a design that is more Mention-friendly, or I can design it in the future.

Also, I will save peoples from the complexity of DbConfig.php as @ardaabi1 said. I'll commit shortly.

**Comment by slaweally** [comment]
Yes, yes, yes, I just gave an idea.

**Comment by par274** [comment]
In the new version you can simply edit the `.env.example` file and rename to `.env` or create an automatic file with the command line. (Of course this will be included in the install)

```bash
php bin/console mention:install-db [--dbadapter DBADAPTER] [--user USER] [--password PASSWORD] [--dbname DBNAME] [--dbhost DBHOST]
```
*Excluding square brackets

OR
Rename:
> `.env.example` to `.env`

Make your database settings in this file.

Or if you using Plesk, make it easy: https://github.com/mentionbb/mentionbb-plesk-setup

**Comment by slaweally** [comment]
> In the new version you can simply edit the `.env.example` file and rename to `.env` or create an automatic file with the command line. (Of course this will be included in the install)
> 
> ```shell
> php bin/console mention:install-db [--dbadapter DBADAPTER] [--user USER] [--password PASSWORD] [--dbname DBNAME] [--dbhost DBHOST]
> ```
> 
> *Excluding square brackets
> 
> OR Rename:
> 
> > `.env.example` to `.env`
> 
> Make your database settings in this file.
> 
> Or if you using Plesk, make it easy: https://github.com/mentionbb/mentionbb-plesk-setup

I think it's better to share release notes for new developments separately. This way it makes more sense than writing it here because users should see it too.

Also the progress is going very fast and good :)

**Comment by par274** [comment]
> > In the new version you can simply edit the `.env.example` file and rename to `.env` or create an automatic file with the command line. (Of course this will be included in the install)
> > ```shell
> > php bin/console mention:install-db [--dbadapter DBADAPTER] [--user USER] [--password PASSWORD] [--dbname DBNAME] [--dbhost DBHOST]
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > *Excluding square brackets
> > OR Rename:
> > > `.env.example` to `.env`
> > 
> > 
> > Make your database settings in this file.
> > Or if you using Plesk, make it easy: https://github.com/mentionbb/mentionbb-plesk-setup
> 
> I think it's better to share release notes for new developments separately. This way it makes more sense than writing it here because users should see it too.
> 
> Also the progress is going very fast and good :)

Actually, they do exist. 

https://github.com/mentionbb/mentionbb/releases/tag/mentionbb-11020000-5
